config/config.yaml

```diff
@@ -39,10 +39,17 @@ storage:
   performance:
     # Number of parallel threads used for search operations. If 0 - auto selection.
     max_search_threads: 0
-    # Max total number of threads, which can be used for running optimization processes across all collections.
-    # Note: Each optimization thread will also use `max_indexing_threads` for index building.
-    # So total number of threads used for optimization will be `max_optimization_threads * max_indexing_threads`
-    max_optimization_threads: 1
+
+    # Max number of threads (jobs) for running optimizations across all collections, each thread runs one job.
+    # If 0 - have no limit and choose dynamically to saturate CPU.
+    # Note: each optimization job will also use `max_indexing_threads` threads by itself for index building.
+    max_optimization_threads: 0
+
+    # CPU budget, how many CPUs (threads) to allocate for an optimization job.
+    # If 0 - auto selection, keep 1 or more CPUs unallocated depending on CPU size
+    # If negative - subtract this number of CPUs from the available CPUs.
+    # If positive - use this exact number of CPUs.
+    cpu_budget: 0
```

Comment:
Should we rename this one to `optimizer_cpu_budget` as well? I don't want to make names more difficult if not necessary.

cc @coszio

Reply:
I agree with you, looks too verbose for being user-facing. Talking about _necessity_, though, the only thing capturing the semantic of this being applied only to optimizers is the comment above it. Not sure either

Reply:
> (...) the only thing capturing the semantic of this being applied only to optimizers is the comment above it. Not sure either

A CPU budget doesn't currently make sense for anything else. But the average user probably doesn't have this Insight.

Reply:
I would vote for `optimizer_cpu_budget`